pkg/virtctl/convert/guess.go

```diff
@@ -0,0 +1,36 @@
+package convert
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"k8s.io/client-go/pkg/util/yaml"
+	"unicode"
+)
+
+func GuessStreamType(source io.Reader, size int) (io.Reader, Type) {
```

Comment:
This function signature should probably also include an error code. For example there might be parse errors, or the stream doesn't match anything we know.